### PR TITLE
IfElseFlattener: Do not de-else or swap if both branches terminate.

### PR DESCRIPTION
**Pull request by ltfish**

### CODE REVIEW

File: tests/analyses/decompiler/test_decompiler.py [hunk 1]
```diff
@@ -2898,6 +2898,131 @@ def test_multistatementexpression_od_read_char(self, decompiler_options=None):
...
+        f = proj.kb.functions["bridge_print_opt"]
+        proj.analyses.CompleteCallingConventions(cfg=cfg, recover_variables=True, analyze_callsites=True)
+        d = proj.analyses[Decompiler](f, cfg=cfg.model, options=decompiler_options)
+
+        self._print_decompilation_result(d)
+        text = d.codegen.text
+        good_if_return = "if (!a2)\n        return 0;\n"
+        first_if_location = text.find("if")
+
+        # TODO: this is broken right now on the 1 goto for a bad else. It may not be relevant for this testcase.
```
Comment (mahaloz) [inline tests/analyses/decompiler/test_decompiler.py:2916]: If you think this is out of scope, we can just remove it, since it may not have anything to do with if-flipping

File: tests/analyses/decompiler/test_decompiler.py [hunk 2]
```diff
@@ -2898,6 +2898,131 @@ def test_multistatementexpression_od_read_char(self, decompiler_options=None):
...
+        # In the correct output, there should only ever be 2 scopes (the function, and a single if-scope) of
+        # deepness in the full function. To verify this, we check that no scope of 3 deepness exists.
+
+        scope_prefix = "    "
+        bad_scope_prefix = scope_prefix * 3
+
+        assert scope_prefix in text
+        assert bad_scope_prefix not in text
+
+        # TODO: fix me, this is a real bug
```
Comment (mahaloz) [inline tests/analyses/decompiler/test_decompiler.py:3020]: This TODO is real and is an if-stmt flipping bug. See the last if-stmt in the if-else chain.

**Comment by mahaloz** [comment]
@ltfish I'm going to move the two unresolved conversations to an issue, then merge this if you have no qualms. CI passes with new test cases (all with expected results).

**Comment by ltfish** [comment]
Oh uh give me a chance to fix these bugs before merging. I have cases to trigger them.

**Comment by ltfish** [comment]
I'll hit merge as soon as the CI passes.
